File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/inventory/exchanges/ExchangeBuyMessage.scala





















// Generated on 05/03/2013 10:34:53
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait ExchangeBuyMessage extends MessageDeserializer {
  val messageId = 5774

  protected trait Message extends BaseMessage with Serializable {
    val messageId = ExchangeBuyMessage.messageId

    val objectToBuyId: Int
    val quantity: Int

    def serialize(buf: Buffer) {
      buf.writeInt(objectToBuyId)
      buf.writeInt(quantity)
    }

  }

  protected def create(objectToBuyId: Int, quantity: Int): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val objectToBuyId = buf.readInt()
    val quantity = buf.readInt()

    create(objectToBuyId, quantity)
  }

}

object ExchangeBuyMessage extends ExchangeBuyMessage {
  case class MessageImpl(objectToBuyId: Int, quantity: Int) extends Message
  type Target = MessageImpl

  override def create(objectToBuyId: Int, quantity: Int) = MessageImpl(objectToBuyId, quantity)

  def apply(objectToBuyId: Int, quantity: Int) = create(objectToBuyId, quantity)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/ExchangeBuyMessage.java


// Generated on 05/08/2013 19:37:53
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class ExchangeBuyMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5774;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int objectToBuyId;
    public int quantity;
    
    public ExchangeBuyMessage() { }
    
    public ExchangeBuyMessage(int objectToBuyId, int quantity) {
        this.objectToBuyId = objectToBuyId;
        this.quantity = quantity;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeInt(objectToBuyId);
        buf.writeInt(quantity);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        objectToBuyId = buf.readInt();
        if (objectToBuyId < 0)
            throw new RuntimeException("Forbidden value on objectToBuyId = " + objectToBuyId + ", it doesn't respect the following condition : objectToBuyId < 0");
        quantity = buf.readInt();
        if (quantity < 0)
            throw new RuntimeException("Forbidden value on quantity = " + quantity + ", it doesn't respect the following condition : quantity < 0");
    }
    
}
